simulation/traffic_simulator/include/traffic_simulator/hdmap_utils/cache.hpp

```diff
@@ -0,0 +1,136 @@
+// Copyright 2015-2020 Tier IV, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TRAFFIC_SIMULATOR__HDMAP_UTILS__CACHE_HPP_
+#define TRAFFIC_SIMULATOR__HDMAP_UTILS__CACHE_HPP_
+
+#include <boost/optional.hpp>
+#include <geometry_msgs/msg/point.hpp>
+#include <mutex>
+#include <scenario_simulator_exception/exception.hpp>
+#include <traffic_simulator/math/catmull_rom_spline.hpp>
+#include <unordered_map>
+#include <vector>
+
+namespace hdmap_utils
+{
+class RouteCache
+{
+public:
+  bool exists(std::int64_t from, std::int64_t to)
+  {
+    std::lock_guard<std::mutex> lock(mutex_);
+    std::pair<std::int64_t, std::int64_t> key;
+    key.first = from;
+    key.second = to;
+    if (data_.find(key) == data_.end()) {
+      return false;
+    }
+    return true;
+  }
+  std::vector<std::int64_t> getRoute(std::int64_t from, std::int64_t to)
+  {
+    if (!exists(from, to)) {
+      THROW_SIMULATION_ERROR(
+        "route from : ", from, " to : ", to, " does not exists on route chache.");
```

Comment:
Capitalize the beginning of the English sentence.
The period at the end of the sentence is duplicated because struct SimulationError is automatically added.

Reply:
This problem should be fixed in all sentence, so I publish it as new issue.
https://github.com/tier4/scenario_simulator_v2/issues/337